tests/test_flow/test_logging.py

```diff
@@ -43,40 +42,62 @@ def x_plus_two(x_plus_one):
 
     flow = builder.build()
     assert flow.get("x_plus_one") == 2
-
     log_checker.expect(
         "Accessed   x(x=1) from definition",
         "Computing  x_plus_one(x=1) ...",
         "Computed   x_plus_one(x=1)",
     )
 
     assert flow.get("x_plus_two") == 3
-    log_checker.expect(
-        "Accessed   x_plus_one(x=1) from in-memory cache",
-        "Computing  x_plus_two(x=1) ...",
-        "Computed   x_plus_two(x=1)",
-    )
+
+    if parallel_processing_enabled:
+        # This is different from serial processing because we don't pass
+        # in-memory cache to the subprocesses. The subprocess loads the
+        # entities from disk cache instead.
+        log_checker.expect(
+            "Loaded     x_plus_one(x=1) from disk cache",
+            "Computing  x_plus_two(x=1) ...",
+            "Computed   x_plus_two(x=1)",
+        )
+    else:
+        log_checker.expect(
+            "Accessed   x_plus_one(x=1) from in-memory cache",
+            "Computing  x_plus_two(x=1) ...",
+            "Computed   x_plus_two(x=1)",
+        )
 
     flow = builder.build()
     assert flow.get("x_plus_one") == 2
-    log_checker.expect(
+    if parallel_processing_enabled:
+        # This is different from serial processing where we access x(x=1)
+        # from definition. We don't need to access it unless we are going
+        # to compute x_plus_one. To be accurate, we do access it for
```

Comment:
nit: it's hard to tell if "to be accurate, we do access it" here means "we access it in order to be accurate" or "To clarify: we do access it".

More generally, I'm finding it hard to understand this paragraph. It doesn't seem to say anything about what specifically about parallel processing is causing this difference.

Reply:
Reworded it, hopefully it's clearer now.